src/lib/Colors.ts

```diff
@@ -0,0 +1,233 @@
+
+	/**
+	 * The text color. Can be a basic string like "red", a hex string, a RGB array, or HSL array.
+	 * @since 0.0.1
+	 */
+	text?: Color;
+}
+
+/**
+ * The format data for Color.
+ * @since 0.0.1
+ */
+export interface ColorsFormatData {
+	/**
+	 * The opening format data styles.
+	 * @since 0.0.1
+	 */
+	opening?: number[];
+	/**
+	 * The closing format data styles.
+	 * @since 0.0.1
+	 */
+	closing?: number[];
+}
+
+/**
+ * The color to apply.
+ * @since 0.0.1
+ */
+export type Color = 'black' | 'green' | 'yellow' | 'blue' | 'red' | 'magenta' | 'cyan' | 'gray' |
+'grey' | 'lightgray' | 'lightgrey' | 'lightred' | 'lightgreen' |
+'lightyellow' | 'lightblue' | 'lightmagenta' | 'lightcyan' | 'white';
+
+/**
+ * The style or styles to apply.
+ * @since 0.0.1
+ */
+export type Style = 'normal' | 'bold' | 'dim' | 'italic' | 'underline' | 'inverse' | 'hidden' | 'strikethrough';
+
+/**
+ * The Colors class that manages the colors displayed in the console.
+ * @since 0.0.1
+ */
+export class Colors {
+
+	/**
+	 * The opening tags.
+	 * @since 0.0.1
+	 */
+	public readonly opening: string;
+
+	/**
+	 * the closing tags.
+	 * @since 0.0.1
+	 */
+	public readonly closing: string;
+
+	/**
+	 * @since 0.0.1
+	 * @param options The options.
+	 * @example
+	 * const red = new Colors({ text: 'red' });
+	 * console.log(red.format('Hello World'));
+	 * // A 'Hello World' string will be printed red.
+	 */
+	public constructor(options: ColorsFormatOptions = {}) {
+		const { opening, closing } = Colors.text(options.text, (this.constructor as typeof Colors).background(options.background, (this.constructor as typeof Colors).style(options.style)));
+
+		this.opening = (this.constructor as typeof Colors).useColors && opening ? `\u001B[${opening.join(';')}m` : '';
+		this.closing = (this.constructor as typeof Colors).useColors && closing ? `\u001B[${closing.join(';')}m` : '';
+	}
+
+	/**
+	 * Format a string.
+	 * @since 0.0.1
+	 * @param string The string to format
+	 */
+	public format(string: string): string {
+		return this.opening + string + this.closing;
+	}
+
+	/**
+	 * Applies the style.
+	 * @since 0.0.1
+	 * @param styles The style or styles to apply
+	 * @param FormatData the format data
+	 */
+	private static style(styles?: Style | Style[], { opening = [], closing = [] }: ColorsFormatData = {}): ColorsFormatData {
+		if (styles) {
+			if (!Array.isArray(styles)) styles = [styles];
+			for (let style of styles) {
+				style = style.toLowerCase() as Style;
+				if (!(style in this.STYLES)) continue;
+				opening.push(this.STYLES[style]);
+				closing.push(this.CLOSE[style]);
+			}
+		}
+		return { opening, closing };
+	}
+
+	/**
+	 * Applies the background.
+	 * @since 0.0.1
+	 * @param background the background to apply
+	 * @param FormatData the format data
+	 */
+	private static background(background?: Color, { opening = [], closing = [] }: ColorsFormatData = {}): ColorsFormatData {
+		if (background && background.toLowerCase() in this.BACKGROUNDS) {
```

Comment:
Let's use `Reflect.has` like everywhere else
```suggestion
		if (background && Reflect.has(this.BACKGROUNDS, background.toLowerCase())) {
```